src/agent/rustjail/src/cgroups/fs/mod.rs

```diff
@@ -313,47 +321,45 @@ where
 {
     let p = format!("{}/{}", dir, file);
     fs::write(p.as_str(), v.to_string().as_bytes())
-        .chain_err(|| format!("couldn't write to file: {:?}", p))?;
+        .map_err(|e| anyhow!(e).context(format!("couldn't write to file: {:?}", p)))?;
```

Comment:
why map_err here?  why  not using  context directly such as: 

fs::write(p.as_str(), v.to_string().as_bytes()).context(format!("couldn't write to file: {:?}", p))?;

Reply:
didn't work

```
error[E0599]: no method named `context` found for enum `std::result::Result<(), std::io::Error>` in the current scope
   --> rustjail/src/cgroups/fs/mod.rs:324:10
    |
324 |         .context(format!("couldn't write to file: {:?}", p))?;
    |          ^^^^^^^ method not found in `std::result::Result<(), std::io::Error>`
```

Reply:
Hi @devimc 

It seems you should import  anyhow's Context  trait  to use context method such as:
use anyhow::{anyhow, Context, Result};

Reply:
you're right, sorry this is my second patch in rust :disappointed: